staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap_test.go

```
@@ -17,45 +17,474 @@ limitations under the License.
+			configMapName: "foo",
+			fromFile:      []string{"=key=/file1"},
+
+			expectErr: true,
+		},
+		"create_invalid_configmap_literal_key_contains_=": {
+			configMapName: "foo",
+			fromFile:      []string{"=key=value1"},
+
+			expectErr: true,
+		},
+		"create_invalid_configmap_duplicate_key1": {
+			configMapName: "foo",
+			fromLiteral:   []string{"key1=value1", "key1=value2"},
+
+			expectErr: true,
+		},
+		"create_invalid_configmap_no_file": {
+			configMapName: "foo",
+			fromFile:      []string{"key1=/file1"},
+
+			expectErr: true,
+		},
+		"create_invalid_configmap_invalid_literal": {
+			configMapName: "foo",
+			fromLiteral:   []string{"key1value1"},
+
+			expectErr: true,
+		},
+		"create_invalid_configmap_invalid_filepath": {
+			configMapName: "foo",
+			fromFile:      []string{"key1==file1"},
+
+			expectErr: true,
+		},
+		"create_invalid_configmap_too_many_args": {
+			configMapName: "too_many_args",
+			fromFile:      []string{"key1=/file1"},
+			fromEnvFile:   "foo",
+
+			expectErr: true,
+		},
+		"create_invalid_configmap_too_many_args_1": {
+			configMapName: "too_many_args_1",
+			fromLiteral:   []string{"key1=value1"},
+			fromEnvFile:   "foo",
+
+			expectErr: true,
+		},
+	}
+
+	// run all the tests
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			configMapOptions := ConfigMapOptions{
+				Name:           test.configMapName,
+				Type:           test.configMapType,
+				AppendHash:     test.appendHash,
+				FileSources:    test.fromFile,
+				LiteralSources: test.fromLiteral,
+				EnvFileSource:  test.fromEnvFile,
+			}
+
+			if test.setup != nil {
+				if teardown := test.setup(t, &configMapOptions); teardown != nil {
+					defer teardown()
+				}
 			}
-		}),
+
+			configMap, err := configMapOptions.createConfigMap()
+			if !test.expectErr && err != nil {
+				t.Errorf("test %s, unexpected error: %v", name, err)
+			}
+			if test.expectErr && err != nil {
```

OK, so this is my opinion and might be a nit.

When I'm testing if something should return an error and it doesn't, I prefer doing something like
```
if test.expectErr && err == nil  {
    t.Errorf("test %s was expecting an error but no error ocurred")
```
this way, instead of returning the function, I would raise an early error saying "Hey, I was expecting an error but this test failed, as no error was found.

By the logic used today, what happens is: I expect an error, error == nil then instead of returning I follow to the next apiequality.Semantic test and this could be resolved earlier :)